core/types/receipt.go

```diff
@@ -93,6 +97,23 @@ type receiptRLP struct {
 	Logs              []*Log
 }
 
+// Quorum
+
+// storedMPSReceiptRLPWithRevertReason is the storage encoding of a receipt which contains
+// receipts per PSI, including Revert Reason
+type storedMPSReceiptRLPWithRevertReason struct {
```

Comment:
should we put all quorum additions in separate file(s)?
- receipt_revert_reasons.go?

all put all additionnal functions at the end of the file?

Reply:
I added in that place as we had already before the struct for MPS here. I do prefer to move to the end. what would you recommend to move?

about another file, I didn't do it to avoid making it public.

Reply:
sure if we continue to add at the same place in order to have less diff, I have nothing to move.

It may be outside the scope of the PR then, if we do some reorganization of blocs.

I would have put all quorum types at the bottom of the original ones, after line 137.
All new GoQ's functions after the original ones, even if they are not grouped.

An idea would be to add some comments that link to the others functions (I will search if we can do some hypertext links in source code with the IDEs)

From my point of view, I would say from a Java dev perspective, I prefer to have the related interface, type, factories, methods in order. Here there are many private types from l.83 to l.168 that should be moved elsewhere, next from where they are used. Indeed I think it would improve the reading. Yet I understand that it's go-ethereum code that we should not modify, only when needed.

In some project we used [quilt](https://linux.die.net/man/1/quilt), that is a layer of diffs than you apply. Maybe those internals improvement, we could handle with a tool like that, on top of our current process.

Reply:
I refactored the code as we discussed on the call.